### PR TITLE
Executing UAMethod should check user authorization

### PR DESCRIPTION
**Pull request by MarcoSeb**
Actual state of the PR
- [X] rebase with master 
- [X] npm install
- [x] npm test to ensure no failing test
- [X] e2e test added
- [X] refernece issue #896

the addad e2e works if I run it seperatly, but not if I run it with npm test.
Hekp welcome!

### CODE REVIEW

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"use Strict";
```
Comment (erossignon) [inline packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: use **s**trict

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+const opcua = require("node-opcua");
+const should = require("should");
+
+describe("Issue 896: Check Authorization for UAMethods", () => {
```
Comment (erossignon) [inline packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: issue #896 ( add pound sign)

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            }
+        }).bindMethod((inputArguments, context, callback) => {
+            wasExecuted = true;
+            callback(null, {statusCode: opcua.StatusCodes.Good})
+        });
+        await server.start();
+        return client.connect(`opc.tcp://localhost:${port}/UA/NodeOPCUA`);
+    });
+
+    after(() => {
```
Comment (erossignon) [inline packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]:
```` typescript
after(async()=>{
   await client.disconnect();
   await server.shutdown();

});
````

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js (hunk 2 above)
Comment (erossignon) [reply packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: Also add describe for leak detection:
````
// tslint:disable-next-line:no-var-requires
const describe = require("node-opcua-leak-detector").describeWithLeakDetector;
describe("Issue #896: Check Authorization for UAMethods", () => {

```

File: package.json [hunk 4]
```diff
@@ -1,163 +1,163 @@
-{
```
Comment (erossignon) [inline package.json]: Please revert to original package.json...

This version has too many changes ( probably necessarry EOL  or different tab space convention)

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js (hunk 3 above)
Comment (erossignon) [reply packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: This should fix the test

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            userName: "Gandalf",
+            password: "g"
+        });
+        const result = await clientSession.call({
+            methodId: "ns=1;s=doIt",
+            objectId: "ns=1;s=e2e",
+            inputArguments: [],
+        });
+        result.statusCode.should.eql(opcua.StatusCodes.Good);
+        wasExecuted.should.eql(true);
```
Comment (erossignon) [inline packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: Don't forget to close the session.

File: package.json (hunk 4 above)
Comment (MarcoSeb) [reply package.json]: For the future it should be considered to add a .gitattributes file, to have a consistent line ending which is not depending on the contributor.
See: https://docs.github.com/en/free-pro-team@latest/github/using-git/configuring-git-to-handle-line-endings

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js (hunk 1 above)
Comment (MarcoSeb) [reply packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: migrated js => ts

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js (hunk 3 above)
Comment (MarcoSeb) [reply packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: thx

File: packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js (hunk 1 above)
Comment (MarcoSeb) [reply packages/node-opcua-end2end-test/test/end_to_end/test_e2e_issue896_execute_method_access_control.js]: @erossignon Who gonna "Resolve conversation"? Is there any convention?
Should it be always the initiator of the convrsation or can I hit the button, if I am convinced that it is fixed?

**Comment by erossignon** [comment]
Thank you !

**Pull request merged**